epan/dissectors/packet-ippusb.c

```diff
@@ -21,70 +21,421 @@
 #include <epan/to_str.h>
 #include <epan/conversation.h>
 #include <epan/wmem/wmem.h>
-#include "packet-http.h"
-#include "packet-usb.h"
+#include <reassemble.h>
+#include <packet-http.h>
+#include <packet-usb.h>
 #include <stdio.h>
 
+#define HTTP 0
+
 void proto_register_ippusb(void);
 void proto_reg_handoff_ippusb(void);
+static int is_http_header(int first_linelen, const guchar *first_line);
+
+/* Reassemble by default */
+static gboolean global_ippusb_reassemble = TRUE;
+static guint32 ippusb_last_pdu = -1;
 
 static int proto_ippusb = -1;
+static gint ett_ippusb = -1;
+static gint ett_ippusb_as = -1;
+static gint ett_ippusb_attr = -1;
+static gint ett_ippusb_member = -1;
+static gint ett_ippusb_fragment= -1;
+static gint ett_ippusb_fragments = -1;
+
+/* For reassembly */
+static gint hf_ippusb_fragments = -1;
+static gint hf_ippusb_fragment = -1;
+static gint hf_ippusb_fragment_overlap = -1;
+static gint hf_ippusb_fragment_overlap_conflict = -1;
+static gint hf_ippusb_fragment_multiple_tails = -1;
+static gint hf_ippusb_fragment_too_long_fragment = -1;
+static gint hf_ippusb_fragment_error = -1;
+static gint hf_ippusb_fragment_count = -1;
+static gint hf_ippusb_reassembled_in = -1;
+static gint hf_ippusb_reassembled_length = -1;
+static gint hf_ippusb_reassembled_data = -1;
+
+static const fragment_items ippusb_frag_items = {
+  &ett_ippusb_fragment,
+  &ett_ippusb_fragments,
+  &hf_ippusb_fragments,
+  &hf_ippusb_fragment,
+  &hf_ippusb_fragment_overlap,
+  &hf_ippusb_fragment_overlap_conflict,
+  &hf_ippusb_fragment_multiple_tails,
+  &hf_ippusb_fragment_too_long_fragment,
+  &hf_ippusb_fragment_error,
+  &hf_ippusb_fragment_count,
+  &hf_ippusb_reassembled_in,
+  &hf_ippusb_reassembled_length,
+  &hf_ippusb_reassembled_data,
+  "IPPUSB fragments"
+};
+
+struct ippusb_multisegment_pdu {
+	guint32 nxtpdu;
+	guint32 first_frame;
+	guint32 last_frame;
+    guint32 running_size;
+    gboolean finished;
+    guint32 flags;
+};
+
+struct ippusb_multisegment_pdu *
+pdu_store(packet_info *pinfo, wmem_tree_t *multisegment_pdus)
+{
+    struct ippusb_multisegment_pdu *msp;
+
+    msp=wmem_new(wmem_file_scope(), struct ippusb_multisegment_pdu);
+    msp->first_frame=pinfo->num;
+    msp->last_frame=pinfo->num;
+    msp->finished = FALSE;
+    msp->flags=0;
+    wmem_tree_insert32(multisegment_pdus, pinfo->num, (void *)msp);
+
+    return msp;
+}
+
+struct ippusb_analysis {
+
+	wmem_tree_t *multisegment_pdus;
+
+};
+
+static struct ippusb_analysis *
+init_ippusb_conversation_data()
+{
+    struct ippusb_analysis *ippusbd;
+
+    /* Initialize the tcp protocol data structure to add to the tcp conversation */
+    ippusbd=wmem_new0(wmem_file_scope(), struct ippusb_analysis);
```

Comment:
>= [](start = 11, length = 1)

Add spaces on each side to align with convention.

Reply:
Other places below as well. :)

---
In reply to: [454656437](https://github.com/tjamhar/wireshark/pull/2#discussion_r454656437) [](ancestors = 454656437)